arwen/host/execution.go

```diff
@@ -202,25 +202,28 @@ func copyTxHashesFromContext(copyEnabled bool, runtime arwen.RuntimeContext, inp
 func (host *vmHost) ExecuteOnDestContext(input *vmcommon.ContractCallInput) (vmOutput *vmcommon.VMOutput, asyncInfo *arwen.AsyncContextInfo, err error) {
 	log.Trace("ExecuteOnDestContext", "caller", input.CallerAddr, "dest", input.RecipientAddr, "function", input.Function)
 
-	builtinFunctionExecuted := false
 	scExecutionInput := input
 
+	blockchain := host.Blockchain()
```

Comment:
we have to delete any user of RevertESDT - there are 2 public functions. those must deleted

Reply:
you have to remove host.RevertESDTTransfer(input) from TransferAndExecute function. Now you revert it through snapshot. after that you send tokens again.

Reply:
the same happens on asnycCall with revert -> you make a revert snasphot here, but you make and actual transfer later - which is clearly wrong.